src/amp-story-player/amp-story-player-impl.js

```diff
@@ -597,6 +606,24 @@ export class AmpStoryPlayer {
     });
   }
 
+  /**
+   * Updates the controls config for a given story.
+   * @param {number} storyIdx
+   * @private
+   */
+  updateControlsStateForStory_(storyIdx) {
+    // Disables skip-next-button when story is the last one in the player.
+    if (storyIdx === this.stories_.length - 1) {
+      const skipButtonIdx = this.playerConfig_.controls.findIndex(
+        (control) => control.name === 'skip-next-button'
+      );
+
+      if (skipButtonIdx >= 0) {
+        this.playerConfig_.controls[skipButtonIdx].state = 'disabled';
```

Comment:
This will "Update the controls" for all the stories. I know in your case it's fine because that's the last story, but if you re-use this method for anything else in the future you might create a bug here.

Two possible solutions, I'll let you decide which one (if any) you think works best here:
1. Make this method "pure": it accepts an input and returns an output, but doesn't change any object level state (here the `controls`)
2. Rename the method to make it clear that it updates the controls for all stories (eg: `updateControlsStateForAllStoriesYesEventTheNextOnes_`)